Source/WebCore/xml/XMLHttpRequest.h

```diff
@@ -54,13 +55,15 @@ class XMLHttpRequestUpload;
 struct OwnedString;
 template<typename> class ExceptionOr;
 
-class XMLHttpRequest final : public ActiveDOMObject, public RefCounted<XMLHttpRequest>, private ThreadableLoaderClient, public XMLHttpRequestEventTarget {
+class XMLHttpRequest final : public ActiveDOMObject, public RefCountedAndCanMakeWeakPtr<XMLHttpRequest>, private ThreadableLoaderClient, public XMLHttpRequestEventTarget {
     WTF_MAKE_TZONE_OR_ISO_ALLOCATED_EXPORT(XMLHttpRequest, WEBCORE_EXPORT);
     WTF_OVERRIDE_DELETE_FOR_CHECKED_PTR(XMLHttpRequest);
 public:
     void ref() const final { RefCounted::ref(); }
     void deref() const final { RefCounted::deref(); }
 
+    USING_CAN_MAKE_WEAKPTR(CanMakeWeakPtr<XMLHttpRequest>);
```

Comment:
Not needed. This should have been your hint that you are doing something odd.

Reply:
I have added this back, but now with `EventTarget` as argument.